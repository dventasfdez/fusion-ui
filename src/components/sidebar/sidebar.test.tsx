import React from "react";
import renderer from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react";
import Sidebar, { ISidebarProps, SidebarButton, SidebarDivider, SidebarLogo } from "./sidebar";

const sidebarExample = (props?: ISidebarProps | any) => (
  <Sidebar data-testid="sidebar" {...props} className="stepone-ui">
    <SidebarLogo data-testid="sidebar-logo">
      <img
        src="https://www-prd-amz930-com.azureedge.net/es-es/-/media/project/adeccogroup/horizontal-the-adecco-group-brand-mark-land-rgb.png?h=475&w=1385&modified=00010101000000&hash=950A28C29AD61BBB9CB374DA671A06D1"
        alt="logo"
      />
    </SidebarLogo>
    <SidebarButton
      disabled={props?.buttonDisabled ? props.buttonDisabled : false}
      dropdown={props?.dropdown ? props.dropdown : false}
      onClick={props?.onClickButton ? props.onClickButton : null}
      data-testid="sidebar-button-1"
      id="button1"
      icon={<span className="material-icons">home</span>}
      label="Home"
      href="#1"
    />
    <SidebarButton data-testid="sidebar-button-2" id="button2" icon={<span className="material-icons">home</span>} label="Home" href="#2" />
    <SidebarButton data-testid="sidebar-button-3" id="button3" icon={<span className="material-icons">home</span>} label="Home" href="#3" />
    <SidebarButton data-testid="sidebar-button-4" id="button4" disabled icon={<span className="material-icons">home</span>} label="Home" href="#4" />
    <SidebarDivider data-testid="sidebar-divider">divider text</SidebarDivider>
    <SidebarButton data-testid="sidebar-button-5" id="button5" icon={<span className="material-icons">home</span>} label="Home" href="#5" />
    <SidebarButton data-testid="sidebar-button-6" id="button6" icon={<span className="material-icons">home</span>} label="Home" href="#6" />
    <SidebarButton data-testid="sidebar-button-7" id="button7" icon={<span className="material-icons">home</span>} label="Home" href="#7" />
    <SidebarButton data-testid="sidebar-button-8" id="button8" icon={<span className="material-icons">home</span>} label="Home" href="#8" />
    <SidebarButton data-testid="sidebar-button-9" id="button9" icon={<span className="material-icons">home</span>} label="Home" href="#12" />
    <SidebarButton data-testid="sidebar-button-10" id="button10" icon={<span className="material-icons">home</span>} label="Home" href="#123" />
    <SidebarButton data-testid="sidebar-button-11" id="button11" icon={<span className="material-icons">home</span>} label="Home" href="#145" />
  </Sidebar>
);

it("Sidebar should render", () => {
  const component = renderer.create(sidebarExample());
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("Sidebar collapsed should render", () => {
  const component = renderer.create(sidebarExample());
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("Sidebar default item should render", () => {
  const component = renderer.create(sidebarExample({ defaultShow: true }));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("Click in sidebar item ", () => {
  const { getByTestId } = render(sidebarExample());

  const itemBtn = getByTestId("sidebar-button-2");

  if (itemBtn) {
    fireEvent.click(itemBtn);
    expect(itemBtn.className.trim()).toBe("sidebar-button_selected");
  }
});

it("Click in collapsed btn ", () => {
  const { container, getByTestId } = render(sidebarExample());

  const itemBtn = getByTestId("sidebar-collapsed-btn");
  expect(container.getElementsByClassName("sidebar").length).toBe(1);
  if (itemBtn) {
    fireEvent.click(itemBtn);
  }
  expect(container.getElementsByClassName("sidebar_collapsed").length).toBe(1);
});

it("Custom onClick", () => {
  const onClick = jest.fn();
  const { getByTestId } = render(sidebarExample({ onClick }));

  const itemBtn = getByTestId("sidebar-button-2");

  if (itemBtn) {
    fireEvent.click(itemBtn);
    expect(onClick).toHaveBeenCalled();
  }
});

it("Wrong custom onClick", () => {
  const onClick = jest.fn();
  const { getByTestId } = render(sidebarExample({ onClickButton: onClick }));

  const itemBtn = getByTestId("sidebar-button-1");

  if (itemBtn) {
    fireEvent.click(itemBtn);
    expect(onClick).toHaveBeenCalled();
  }
});

it("Sidebar with dropdown", () => {
  const component = renderer.create(sidebarExample({ dropdown: true }));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("Sidebar with selected dropdown and disabled", () => {
  const component = renderer.create(sidebarExample({ dropdown: true, buttonDisabled: true, defaultItemSelected: "button1" }));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("Render empty sidebar", () => {
  const component = renderer.create(<Sidebar />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("Render Sidebar with unique child", () => {
  const component = renderer.create(
    <Sidebar>
      <div>
        <img
          src="https://www-prd-amz930-com.azureedge.net/es-es/-/media/project/adeccogroup/horizontal-the-adecco-group-brand-mark-land-rgb.png?h=475&w=1385&modified=00010101000000&hash=950A28C29AD61BBB9CB374DA671A06D1"
          alt="logo"
        />
      </div>
    </Sidebar>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("Render sidebar with onClickSidebarItem and click in item", () => {
  const onClickSidebarItem = jest.fn();
  const { getByTestId } = render(sidebarExample({ onClickSidebarItem }));

  const itemBtn = getByTestId("sidebar-button-2");

  if (itemBtn) {
    fireEvent.click(itemBtn);
    expect(onClickSidebarItem).toHaveBeenCalled();
  }
});

it("Render Sidebar and rerender with new props", () => {
  const { rerender, getByTestId } = render(sidebarExample({ defaultItemSelected: "button1" }));
  const itemBtn = getByTestId("sidebar-button-1");
  expect(itemBtn.className.trim()).toBe("sidebar-button_selected");
  rerender(sidebarExample({ defaultItemSelected: "button2" }));
  const itemBtn2 = getByTestId("sidebar-button-2");
  expect(itemBtn2.className.trim()).toBe("sidebar-button_selected");
});

it("Render SIdebar and handle collapse with custom onCollapse", () => {
  const onCollapse = jest.fn();
  const { getByTestId } = render(sidebarExample({ defaultShow: true, onCollapse }));
  const itemBtn = getByTestId("sidebar-collapsed-btn");
  expect(itemBtn.className.trim()).toBe("sidebar-collapsed-button");
  fireEvent.click(itemBtn);
  expect(itemBtn.className.trim()).toBe("sidebar-collapsed-button");
});

it("render with wrong components", () => {
  const { getByTestId } = render(
    <Sidebar data-testid="sidebar">
      <div>Wrong 1</div>
      <div>
        <div>Wrong with child 1</div>
      </div>
    </Sidebar>
  );
  const itemBtn = getByTestId("sidebar");
  expect(itemBtn.className.trim()).toBe("sidebar");
});
