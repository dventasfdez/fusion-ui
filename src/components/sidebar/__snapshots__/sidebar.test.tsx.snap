// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Sidebar with unique child 1`] = `
<div
  className="sidebar "
>
  <button
    className="sidebar-collapsed-button"
    onClick={[Function]}
    type="button"
  >
    <span
      className="material-icons"
    >
      menu_open
    </span>
  </button>
  <div>
    <img
      alt="logo"
      src="https://www-prd-amz930-com.azureedge.net/es-es/-/media/project/adeccogroup/horizontal-the-adecco-group-brand-mark-land-rgb.png?h=475&w=1385&modified=00010101000000&hash=950A28C29AD61BBB9CB374DA671A06D1"
    />
  </div>
</div>
`;

exports[`Render empty sidebar 1`] = `
<div
  className="sidebar "
>
  <button
    className="sidebar-collapsed-button"
    onClick={[Function]}
    type="button"
  >
    <span
      className="material-icons"
    >
      menu_open
    </span>
  </button>
</div>
`;

exports[`Sidebar collapsed should render 1`] = `
<div
  className="sidebar.stepone-ui"
  data-testid="sidebar"
>
  <div
    className="sidebar-logo "
    data-testid="sidebar-logo"
  >
    <img
      alt="logo"
      src="https://www-prd-amz930-com.azureedge.net/es-es/-/media/project/adeccogroup/horizontal-the-adecco-group-brand-mark-land-rgb.png?h=475&w=1385&modified=00010101000000&hash=950A28C29AD61BBB9CB374DA671A06D1"
    />
  </div>
  <button
    className="sidebar-collapsed-button"
    data-testid="sidebar-collapsed-btn"
    onClick={[Function]}
    type="button"
  >
    <span
      className="material-icons"
    >
      menu_open
    </span>
  </button>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-1"
    href="#1"
    id="button1"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-2"
    href="#2"
    id="button2"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-3"
    href="#3"
    id="button3"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button_disabled "
    data-testid="sidebar-button-4"
    href="#4"
    id="button4"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <div
    className="sidebar-divider "
    data-testid="sidebar-divider"
  >
    divider text
  </div>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-5"
    href="#5"
    id="button5"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-6"
    href="#6"
    id="button6"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-7"
    href="#7"
    id="button7"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-8"
    href="#8"
    id="button8"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-9"
    href="#12"
    id="button9"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-10"
    href="#123"
    id="button10"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-11"
    href="#145"
    id="button11"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
</div>
`;

exports[`Sidebar default item should render 1`] = `
<div
  className="sidebar.stepone-ui"
  data-testid="sidebar"
  defaultShow={true}
>
  <div
    className="sidebar-logo "
    data-testid="sidebar-logo"
  >
    <img
      alt="logo"
      src="https://www-prd-amz930-com.azureedge.net/es-es/-/media/project/adeccogroup/horizontal-the-adecco-group-brand-mark-land-rgb.png?h=475&w=1385&modified=00010101000000&hash=950A28C29AD61BBB9CB374DA671A06D1"
    />
  </div>
  <button
    className="sidebar-collapsed-button"
    data-testid="sidebar-collapsed-btn"
    onClick={[Function]}
    type="button"
  >
    <span
      className="material-icons"
    >
      menu_open
    </span>
  </button>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-1"
    href="#1"
    id="button1"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-2"
    href="#2"
    id="button2"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-3"
    href="#3"
    id="button3"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button_disabled "
    data-testid="sidebar-button-4"
    href="#4"
    id="button4"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <div
    className="sidebar-divider "
    data-testid="sidebar-divider"
  >
    divider text
  </div>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-5"
    href="#5"
    id="button5"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-6"
    href="#6"
    id="button6"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-7"
    href="#7"
    id="button7"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-8"
    href="#8"
    id="button8"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-9"
    href="#12"
    id="button9"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-10"
    href="#123"
    id="button10"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-11"
    href="#145"
    id="button11"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
</div>
`;

exports[`Sidebar should render 1`] = `
<div
  className="sidebar.stepone-ui"
  data-testid="sidebar"
>
  <div
    className="sidebar-logo "
    data-testid="sidebar-logo"
  >
    <img
      alt="logo"
      src="https://www-prd-amz930-com.azureedge.net/es-es/-/media/project/adeccogroup/horizontal-the-adecco-group-brand-mark-land-rgb.png?h=475&w=1385&modified=00010101000000&hash=950A28C29AD61BBB9CB374DA671A06D1"
    />
  </div>
  <button
    className="sidebar-collapsed-button"
    data-testid="sidebar-collapsed-btn"
    onClick={[Function]}
    type="button"
  >
    <span
      className="material-icons"
    >
      menu_open
    </span>
  </button>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-1"
    href="#1"
    id="button1"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-2"
    href="#2"
    id="button2"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-3"
    href="#3"
    id="button3"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button_disabled "
    data-testid="sidebar-button-4"
    href="#4"
    id="button4"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <div
    className="sidebar-divider "
    data-testid="sidebar-divider"
  >
    divider text
  </div>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-5"
    href="#5"
    id="button5"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-6"
    href="#6"
    id="button6"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-7"
    href="#7"
    id="button7"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-8"
    href="#8"
    id="button8"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-9"
    href="#12"
    id="button9"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-10"
    href="#123"
    id="button10"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-11"
    href="#145"
    id="button11"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
</div>
`;

exports[`Sidebar with dropdown 1`] = `
<div
  className="sidebar.stepone-ui"
  data-testid="sidebar"
  dropdown={true}
>
  <div
    className="sidebar-logo "
    data-testid="sidebar-logo"
  >
    <img
      alt="logo"
      src="https://www-prd-amz930-com.azureedge.net/es-es/-/media/project/adeccogroup/horizontal-the-adecco-group-brand-mark-land-rgb.png?h=475&w=1385&modified=00010101000000&hash=950A28C29AD61BBB9CB374DA671A06D1"
    />
  </div>
  <button
    className="sidebar-collapsed-button"
    data-testid="sidebar-collapsed-btn"
    onClick={[Function]}
    type="button"
  >
    <span
      className="material-icons"
    >
      menu_open
    </span>
  </button>
  <div
    className="dropdown sidebar-button_dropdown"
    data-show={false}
  >
    <div
      className="dropdown-button "
      onClick={[Function]}
    >
      <div
        className="sidebar-button "
        data-testid="sidebar-button-1"
        id="button1"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
          home
        </span>
        Home
        <span
          className="material-icons sidebar-button-arrow"
        >
          arrow_right
        </span>
      </div>
    </div>
  </div>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-2"
    href="#2"
    id="button2"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-3"
    href="#3"
    id="button3"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button_disabled "
    data-testid="sidebar-button-4"
    href="#4"
    id="button4"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <div
    className="sidebar-divider "
    data-testid="sidebar-divider"
  >
    divider text
  </div>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-5"
    href="#5"
    id="button5"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-6"
    href="#6"
    id="button6"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-7"
    href="#7"
    id="button7"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-8"
    href="#8"
    id="button8"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-9"
    href="#12"
    id="button9"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-10"
    href="#123"
    id="button10"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-11"
    href="#145"
    id="button11"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
</div>
`;

exports[`Sidebar with selected dropdown and disabled 1`] = `
<div
  buttonDisabled={true}
  className="sidebar.stepone-ui"
  data-testid="sidebar"
  dropdown={true}
>
  <div
    className="sidebar-logo "
    data-testid="sidebar-logo"
  >
    <img
      alt="logo"
      src="https://www-prd-amz930-com.azureedge.net/es-es/-/media/project/adeccogroup/horizontal-the-adecco-group-brand-mark-land-rgb.png?h=475&w=1385&modified=00010101000000&hash=950A28C29AD61BBB9CB374DA671A06D1"
    />
  </div>
  <button
    className="sidebar-collapsed-button"
    data-testid="sidebar-collapsed-btn"
    onClick={[Function]}
    type="button"
  >
    <span
      className="material-icons"
    >
      menu_open
    </span>
  </button>
  <div
    className="dropdown sidebar-button_dropdown"
    data-show={false}
  >
    <div
      className="dropdown-button "
      onClick={[Function]}
    >
      <div
        className="sidebar-button_selected_disabled "
        data-testid="sidebar-button-1"
        id="button1"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
          home
        </span>
        Home
        <span
          className="material-icons sidebar-button-arrow"
        >
          arrow_right
        </span>
      </div>
    </div>
  </div>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-2"
    href="#2"
    id="button2"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-3"
    href="#3"
    id="button3"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button_disabled "
    data-testid="sidebar-button-4"
    href="#4"
    id="button4"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <div
    className="sidebar-divider "
    data-testid="sidebar-divider"
  >
    divider text
  </div>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-5"
    href="#5"
    id="button5"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-6"
    href="#6"
    id="button6"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-7"
    href="#7"
    id="button7"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-8"
    href="#8"
    id="button8"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-9"
    href="#12"
    id="button9"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-10"
    href="#123"
    id="button10"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
  <a
    className="sidebar-button "
    data-testid="sidebar-button-11"
    href="#145"
    id="button11"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      home
    </span>
    <span>
      Home
    </span>
  </a>
</div>
`;
